internal/provider/provider.go

```diff
@@ -14,6 +14,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-log/tflog"
 
 	tsClient "github.com/timescale/terraform-provider-timescale/internal/client"
+	d_products "github.com/timescale/terraform-provider-timescale/internal/provider/datasource/products"
```

Comment:
I'm a little conflicted on having the duplicated packages with different import paths. WDYT of this setup

```
provider/
    products/
        data_source.go
    service/
        resource.go
        datasource.go
    vpcs/
         resource.go
         datasource.go
   provider.go
```